dvc/remote/ssh/__init__.py

```diff
@@ -129,20 +133,41 @@ def ssh(self, path_info):
             password=self.password,
         )
 
-    def exists(self, path_infos):
-        single_path = False
+    def exists(self, path_info):
+        with self.ssh(path_info) as ssh:
+            return ssh.exists(path_info.path)
 
-        if not isinstance(path_infos, list):
-            single_path = True
-            path_infos = [path_infos]
+    def batch_exists(self, path_infos, callback):
+        def _exists(chunk_and_channel):
+            chunk, channel = chunk_and_channel
+            ret = []
+            for path in chunk:
+                try:
+                    channel.stat(path)
+                    ret.append(True)
+                except IOError as exc:
+                    if exc.errno != errno.ENOENT:
+                        raise
+                    ret.append(False)
+                callback.update(path)
+            return ret
 
         with self.ssh(path_infos[0]) as ssh:
-            results = [ssh.exists(path_info.path) for path_info in path_infos]
+            with ssh.mute_paramiko():
```

Comment:
If we don't do this, it will log something like: Secsh channel open FAILED: open failed: Administratively prohibited when trying to figure out the max possible connections.

Maybe I should add a comment about this.